daemon/oci_linux.go

```diff
@@ -903,6 +903,18 @@ func (daemon *Daemon) createSpec(c *container.Container) (retSpec *specs.Spec, e
 	s.Process.OOMScoreAdj = &c.HostConfig.OomScoreAdj
 	s.Linux.MountLabel = c.MountLabel
 
+	// Set the masked and readonly paths with regard to the host config options if they are set.
+	if c.HostConfig.MaskedPaths != nil {
+		s.Linux.MaskedPaths = c.HostConfig.MaskedPaths
+	} else {
+		c.HostConfig.MaskedPaths = oci.DefaultSpec().Linux.MaskedPaths // Set it to the default if nil
```

Comment:
Given that `s` is initially set to `oci.DefaultSpec()`, probably could be;

```go
c.HostConfig.MaskedPaths = s.Linux.MaskedPaths
```

I'm a bit unsure though if this is the right location to change this; is it expected that creating a spec will modify the Container-config that was passed? Or should we set these defaults in an earlier stage?

Also looking at where this is called, and it looks to be on [`containerStart()`](https://github.com/moby/moby/blob/6cb75dd5b61f438b026572aca0c191c7d2fb7694/daemon/start.go#L154-L157), which means that inspecting a _created_ container won't show these (and goes a bit against the principle that containers (configuration) should(r)(tm) be immutable).

Perhaps it should be done in [`daemon.containerCreate()`](https://github.com/moby/moby/blob/6cb75dd5b61f438b026572aca0c191c7d2fb7694/daemon/create.go#L32-L81), [`daemon.create()`](https://github.com/moby/moby/blob/6cb75dd5b61f438b026572aca0c191c7d2fb7694/daemon/create.go#L83-L197), or [`createContainerOSSpecificSettings()`](https://github.com/moby/moby/blob/3a633a712c8bbb863fe7e57ec132dd87a9c4eff7/daemon/create_unix.go#L18-L61) (if this is Linux only).


Thinking out loud here; perhaps I'm completely off target here, ha!

Reply:
oh thanks... yeah I am unsure of the best place either...

Reply:
ok I moved and added to the test to ensure the inspect output after create is correct :) and after start I have tests for that as well